.evergreen/resync-specs.sh

```diff
@@ -69,6 +69,9 @@ cpjson () {
 for spec in "$@"
 do
   case "$spec" in
+    atlas-data-lake-testing|data_lake)
```

Comment:
Is there a specific reason to mix underscores and dashes? It would be more consistent with the rest of the script to use `*`

Reply:
My reason was that the * makes it harder to copy paste, eg:
```
.evergreen/resync-specs.sh data*lake      
zsh: no matches found: data*lake
```

Reply:
If the script is like this:
```
  case "$spec" in
    atlas*data*lake)
```
I don't want to type out "atlas-data-lake" manually, instead I want to copy/paste the match condition "atlas*data*lake" which ends up erroring:
```
.evergreen/resync-specs.sh atlas*data*lake      
zsh: no matches found: atlas*data*lake
```

Reply:
```pbpaste | python -c "import sys; s = sys.stdin.read(); print(s.replace('*', '_'));"``` works on my machine. I think it makes more sense to adhere to the style of the rest of the script in the interest of readability and consistency. Conversely, I think that if you would like to make the change to explicit underscores, then perhaps we should open a new ticket to update the rest of the cases as well. I'm also confused what the purpose of having shortened names is if the intended use case is copying from the switch statement. Is it not faster to simply type "data-lake" as opposed to opening the script, finding the right switch statement, copying it, pasting it, doing that for every resync? Another solution is we could add comments above each switch statement with the string ready for copy pasting.

Reply:
The intention here is to be user friendly (which ` pbpaste | python -c "import sys; s = sys.stdin.read(); print(s.replace('*', '_'));"` is mostly certainly not) and match both the spec dir name and the python test dir name (and match common abbreviations like CMAP/SDAM/etc..). I used `*` initially because I was lazy and didn't want to include both `_` and `-` names like `bson_corpus|bson-corpus)` but I now realized this was a mistake. I've updated the script to not use `*`.

Reply:
SGTM